.github/workflows/repro-check.yml

```diff
@@ -0,0 +1,60 @@
+name: Repro check
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      github_sha:
+        description: Commit on which the repro check should be run
+        required: true
+        type: string
+  # Used as a reusable workflow within release-testing workflow
+  workflow_call:
+    inputs:
+      github_sha:
+        description: Commit on which the repro check should be run
+        required: true
+        type: string
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  repro-check:
+    name: Run repro-check for ${{ inputs.github_sha }}
+    runs-on:
+      labels: dind-large
+    container:
+      image: ubuntu:24.04
+      options: >-
+        -e NODE_NAME --privileged --cgroupns host
+    steps:
+      - name: Setup podman
+        shell: bash
+        run: |
+          apt-get update
+          apt-get -y upgrade
+          apt-get -y install curl sudo git podman
+
+          # Create a new non-root user with home directory
+          useradd -m -s /bin/bash ciuser
+          
+          # Create a passwordless sudo rule for the current user (usually root in containers, but safe to include)
+          # Required because the repro check uses it.
+          echo "ciuser ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/nopasswd
+          chmod 0440 /etc/sudoers.d/nopasswd
+
+      # Taken from `ci-main.yml`
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: ${{ github.event_name == 'pull_request' && 256 || 0 }}
```

Comment:
This workflow can't be triggered from a `pull_request` so I would remove this whole fetch-depth line:
```suggestion
```